src/v/cluster/controller_backend.cc

```diff
@@ -582,6 +585,26 @@ ss::future<std::error_code> revert_configuration_update(
       replicas);
     co_return co_await p->update_replica_set(std::move(brokers), rev);
 }
+
+bool is_finishing_operation(
+  const topic_table::delta& operation, const topic_table::delta& candidate) {
+    if (candidate.type != topic_table_delta::op_type::update_finished) {
+        return false;
+    }
+
+    if (operation.new_assignment == candidate.new_assignment) {
+        return true;
+    }
+    if (
+      operation.type == topic_table_delta::op_type::cancel_update
+      || operation.type == topic_table_delta::op_type::force_abort_update) {
+        return operation.previous_replica_set
+               == candidate.new_assignment.replicas;
+    }
```

Comment:
Why do we need to compare assignments? What is the scenario where we see an update_finished delta but still need to execute the preceding update commands?

Reply:
When i think of it now i find it not required, although it was checked before, can we remove this as a follow up ?